drive/src/contract/mod.rs

```diff
@@ -146,6 +153,34 @@ impl Contract {
             .try_into()
             .map_err(|_| Error::CorruptedData(String::from("contract_id must be 32 bytes")))?;
 
+        // Does the contract keep history when the contract itself changes?
+        let keeps_history: bool = bool_for_system_value_from_hash_map(
+            &contract,
+            "$keepsHistory",
+            crate::contract::defaults::DEFAULT_CONTRACT_KEEPS_HISTORY,
+        )?;
+
+        // Is the contract mutable?
+        let mutable: bool = bool_for_system_value_from_hash_map(
+            &contract,
+            "$mutable",
+            crate::contract::defaults::DEFAULT_CONTRACT_MUTABILITY,
+        )?;
+
```

Comment:
We need to make sure that we properly handle changes of these flags during data contract update

Reply:
Done